### PR TITLE
Refactor: Move source directory deletion logic to FileSystem module

### PR DESCRIPTION
**Pull request by manoj-bhaskaran**
## Summary

Extracted the `Remove-SourceDirectory` function and its five private helpers from `Expand-ZipsAndClean.ps1` into a new reusable module in `Core/FileSystem`. This improves code organization, testability, and reusability across scripts that need robust directory deletion with non-zip cleanup capabilities.

## Key Changes

- **Created `Core/FileSystem` module functions** (6 new files):
  - `Remove-SourceDirectory` (public) — orchestrates provider-path resolution, blocking-zip detection, non-zip classification, and robust directory deletion
  - `Get-SourceDirectoryItems` (private) — scans directories recursively with error surfacing
  - `Test-HasBlockingZips` (private) — detects leftover zip files that would be destroyed
  - `Get-NonZipDeletionBlockReason` (private) — classifies deletion blockers (non-zip files vs. empty subdirs)
  - `Remove-NonZipItems` (private) — removes non-zip items depth-first to avoid "directory not empty" errors
  - `Remove-DirectoryRobust` (private) — uses `[System.IO.Directory]::Delete` with `Remove-Item` fallback to work around PowerShell issue #8211 on Linux CI runners

- **Updated `Expand-ZipsAndClean.ps1`**:
  - Removed 172 lines of script-local function definitions
  - Now imports and calls `Remove-SourceDirectory` from the `FileSystem` module
  - Behavior, error messages, and delete semantics remain identical

- **Added comprehensive test suite** (`FileSystem.Tests.ps1`):
  - 11 tests covering public `Remove-SourceDirectory` function
  - 10 tests covering private helper functions via `InModuleScope`
  - Tests validate blocking-zip detection, non-zip classification, nested deletion, error handling, and edge cases

- **Updated module metadata**:
  - `FileSystem.psd1`: version bumped to 1.3.0, description updated
  - `FileSystem.CHANGELOG.md`: documented new functions and behavior
  - `Expand-ZipsAndClean.CHANGELOG.md`: documented refactoring in version 2.6.12

## Implementation Details

- All functions append errors to a mutable `ErrorList` parameter rather than throwing, allowing callers to continue on partial failures
- `Remove-NonZipItems` uses `@(...)` guards on split/filter expressions to preserve correct `.Count` behavior under `Set-StrictMode -Version Latest`
- `Remove-DirectoryRobust` prioritizes `[System.IO.Directory]::Delete` (synchronous .NET primitive) over `Remove-Item -Recurse -Force` to avoid leaving root directories behind on some GitHub Actions Linux runners
- Tests use `[System.IO.Directory]::Exists()` as the source of truth for deletion success, with rich diagnostic output for CI failure investigation

https://claude.ai/code/session_0183NEfFNwHNhJdniXshYMnP

### CODE REVIEW

File: src/powershell/modules/Core/FileSystem/Private/Get-NonZipDeletionBlockReason.ps1 [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    param(
+        [Parameter(Mandatory)][AllowEmptyCollection()][object[]]$NonZips,
+        [Parameter(Mandatory)][bool]$ShouldCleanNonZips,
+        [Parameter(Mandatory)][string]$SourceDir
+    )
+    if ($NonZips.Count -eq 0 -or $ShouldCleanNonZips) { return $null }
+    $hasFiles = @($NonZips | Where-Object { -not $_.PSIsContainer })
+    $msg = ($hasFiles.Count -gt 0) ?
+        "DeleteSource skipped: non-zip files remain in '$SourceDir'. Use -CleanNonZips to remove them." :
+        "DeleteSource skipped: only empty subdirectories remain in '$SourceDir'. Use -CleanNonZips to remove them."
```
[the remark this answers is not in the document]
Comment (manoj-bhaskaran) [reply src/powershell/modules/Core/FileSystem/Private/Get-NonZipDeletionBlockReason.ps1:15]: The module's PowerShellVersion has been raised from 5.1 to 7.0 in FileSystem.psd1. The new private helpers (Get-NonZipDeletionBlockReason, Remove-DirectoryRobust) use the ternary operator ?:, which is PS7-only syntax; the manifest now correctly reflects this. All pre-existing public functions in the module are PS7-compatible. Remove-SourceDirectory also now correctly honours -WhatIf/-Confirm via a ShouldProcess guard that wraps both the non-zip cleanup and the directory deletion.
